server/src/main/java/org/opensearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -105,15 +99,23 @@ public GetIndexResponse(
         this.indices = in.readStringArray();
 
         int mappingsSize = in.readVInt();
-        ImmutableOpenMap.Builder<String, ImmutableOpenMap<String, MappingMetadata>> mappingsMapBuilder = ImmutableOpenMap.builder();
+        ImmutableOpenMap.Builder<String, MappingMetadata> mappingsMapBuilder = ImmutableOpenMap.builder();
         for (int i = 0; i < mappingsSize; i++) {
-            String key = in.readString();
-            int valueSize = in.readVInt();
-            ImmutableOpenMap.Builder<String, MappingMetadata> mappingEntryBuilder = ImmutableOpenMap.builder();
-            for (int j = 0; j < valueSize; j++) {
-                mappingEntryBuilder.put(in.readString(), new MappingMetadata(in));
+            String index = in.readString();
+            if (in.getVersion().before(Version.V_2_0_0)) {
+                int numMappings = in.readVInt();
+                assert numMappings == 0 || numMappings == 1 : "Expected 0 or 1 mappings but got " + numMappings;
+                if (numMappings == 1) {
+                    String type = in.readString();
+                    assert MapperService.SINGLE_MAPPING_NAME.equals(type) : "Expected [_doc] but got [" + type + "]";
+                    mappingsMapBuilder.put(index, new MappingMetadata(in));
+                } else {
+                    mappingsMapBuilder.put(index, MappingMetadata.EMPTY_MAPPINGS);
+                }
+            } else {
+                boolean hasMapping = in.readBoolean();
```

Comment:
You should be able to use `readOptionalWriteable` (and `writeOptionalWriteable`) here, it does manage the mechanics automatically:
```
final MappingMetadata metadata = in.readOptionalWriteable(MappingMetadata::new);
mappingsMapBuilder.put(index, metadata != null ? metadata : MappingMetadata.EMPTY_MAPPINGS);
```

Reply:
done!